app/services/katello/pulp3/repository.rb

```diff
@@ -249,6 +252,22 @@ def ssl_remote_options
         end
       end
 
+      def remote_proxy_url
+        if repo.root.http_proxy_policy == RootRepository::NO_DEFAULT_HTTP_PROXY
+          return nil
+        end
+
+        proxy = repo.root.http_proxy
+        if repo.root.http_proxy_policy == RootRepository::GLOBAL_DEFAULT_HTTP_PROXY
+          proxy = HttpProxy.default_global_content_proxy
+        end
+
+        if proxy
+          return proxy.full_url
+        end
+        nil
```

Comment:
couldn't you just call repo.root.http_proxy&.full_url for this entire method?

Reply:
Did you mean for just the if clause or for the entire `remote_proxy_url` method?

Reply:
the whole method (which probably means we don't actually need it split out as a method